libvast/include/vast/defaults.hpp

```diff
@@ -291,6 +291,14 @@ inline constexpr bool create_partition_index = true;
 /// Whether to spawn central components in separate threads.
 inline constexpr bool detach_components = true;
 
+/// Maximum retries attempts when connecting to a remote vast node.
+inline constexpr std::size_t node_connection_retries = 100u;
+
+/// Time to wait before trying to make another connection attempt to a remote
+/// vast node.
+inline constexpr caf::timespan node_connection_retry_delay
+  = std::chrono::seconds{3u};
```

Comment:
Does it make sense to use an exponential backoff approach here?

E.g., first reconnect takes 20ms, second 40ms, third 80ms, and so on, but it's capped at a configurable maximum value?

We do the same for stalling inputs in readers and there it's been working really well, because for small system hickups you continue almost instantly but for bigger ones you cut the system some slack so you don't overload it.

Also cc @mavam—what do you think?

For the readers we do the following: 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1s, 1s, 1s, ...

Reply:
I don't think so. You'd use exponential backoff mostly when you have multiple parties competing for a resource on a shared medium. Here, we can model this with independent failure event, so a uniform distribution makes the most sense.

I could see—as a follow-up in the future—a dual mode where you try every 1s for 10 times, and then switch to a lower polling frequency. But let's add that only when we need to. Sometimes firewalls are also in the way and temporarily block aggressive connection attempts.